src/torrent.cpp

```diff
@@ -8288,8 +8260,16 @@ namespace libtorrent
 
 		int ret = 0;
 
-		std::int64_t const fin_time = finished_time();
-		std::int64_t const download_time = int(active_time()) - fin_time;
+		const seconds32 act_time = active_time();
+		static_assert(std::is_same<
+			decltype(seed_rank_priority_duration), decltype(act_time)>::value,
+			"Need to be same ratio type to prevent runtime conversion");
+
+		// use raw duration values because active time and finish time use same
+		// ratio type and don't need to be converted to a equal ratio
+		// and we don't compare them agains seconds we just use fraction value
+		std::int32_t const fin_time = finished_time().count();
+		std::int32_t const download_time = act_time.count() - fin_time;
```

Comment:
this code looks suspicious. that static assert does not look right. it seems like turning ``fin_time`` and ``download_time`` into ``duration`` types seems like the most reasonable. that way you get type-safety and nobody needs to stare at this code to be convinced it's correct.

the comparison of ``decltype()`` to make sure the ratio is the same is not clear (and not quite correct). you should just look at the actual ratio, regardless of the other types that go into a duration.